File: selection_sort/select.c

#include<stdio.h>
#include<stdlib.h>

void print_arr(int** arr, int arr_length) {
    printf("[");
    for (int i = 0; i < arr_length; i++) {
        if (i != 0)
            printf(" ");

        printf("%d", i[(*arr)]);
    }
    printf("]\n");

    return;
}

void selectionSort (int **arr, int arr_length) {
    for (int i = 0; i < arr_length; i++) {
        for (int j = i + 1; j < arr_length; j++) {
            if ((*arr)[j] < (*arr)[i]) {
                int temp = (*arr)[i];
                (*arr)[i] = (*arr)[j];
                (*arr)[j] = temp;
            }
        }
    }
        
    return;
}

int main (void) {
    int num_inputs = 0;
    scanf("%d\n", &num_inputs);

    int *inputs = malloc(sizeof(int) * num_inputs);

    for (int i = 0; i < num_inputs; i++) {
        int input = 0;
        scanf("%d\n", &input);

        *(inputs + i) = input;
    }
    
    print_arr(&inputs, num_inputs);

    selectionSort(&inputs, num_inputs);

    print_arr(&inputs, num_inputs);

    return 0;
}

File: selection_sort/select.go
package main

import (
    "fmt"
)

func selectionSort (arr *[]int) {
    for i := 0; i < len(*arr); i++ {
        for j := i + 1; j < len(*arr); j++ {
            if (*arr)[j] < (*arr)[i] {
                temp := (*arr)[i]
                (*arr)[i] = (*arr)[j]
                (*arr)[j] = temp
            }
        }
    }

    return
}

func main () {
    var numInputs int
    fmt.Scanf("%d\n", &numInputs)

    inputs := make([]int, 0, numInputs)
    
    for i := 0; i < numInputs; i++ {
        var input int
        fmt.Scanf("%d\n", &input)

        inputs = append(inputs, input)
    }

    fmt.Printf("%v\n", inputs)
    
    selectionSort(&inputs)

    fmt.Printf("%v\n", inputs)

    return
}
